### PR TITLE
IDEX-3924: Display preview URL in console panel

### PR DESCRIPTION
**Pull request by vzhukovskii**
@slemeur @vparfonov @ashumilova @skabashnyuk

### CODE REVIEW

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/model/impl/CommandImpl.java [hunk 1]
```diff
@@ -65,26 +65,28 @@ public void setType(String type) {
...
+    public void setPreviewUrl(String previewUrl) {
+        this.previewUrl = previewUrl;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        CommandImpl command = (CommandImpl)o;
+        return com.google.common.base.Objects.equal(getName(), command.getName()) &&
```
Comment (azatsarynnyy) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/model/impl/CommandImpl.java]: Why you use ```com.google.common.base.Objects``` instead of ```java.util.Objects```?

Comment (vzhukovskii) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/model/impl/CommandImpl.java]: It's just autoreplacement during generating equals through the Intellij IDEA. Will fix that.

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/model/impl/CommandImpl.java [hunk 2]
```diff
@@ -65,26 +65,28 @@ public void setType(String type) {
...
+    public String getPreviewUrl() {
+        return previewUrl;
+    }
+
+    public void setPreviewUrl(String previewUrl) {
+        this.previewUrl = previewUrl;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (azatsarynnyy) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/model/impl/CommandImpl.java]: This method's body isn't formatted

File: platform-api/che-core-api-core/src/main/java/org/eclipse/che/api/core/model/machine/Command.java [hunk 3]
```diff
@@ -37,4 +37,9 @@
      * Returns command type (i.e. 'maven')
      */
     String getType();
+
+    /**
+     * Returns command preview URL. (i.e. 'http://awesome.com:8080/context')
+     */
+    String getPreviewUrl();
```
Comment (vparfonov) [inline platform-api/che-core-api-core/src/main/java/org/eclipse/che/api/core/model/machine/Command.java]: pls, double check with this changes in API model personally i don't like it. 
But we can accept it if necessary for 4.0-beta 
@skabashnyuk @garagatyi @gazarenkov  your opinion need here

**Comment by TylerJewell** [comment]
1. Can we move the preview URL portion of the command popup to be the last entry, after "Command line"?  The command line is first in people's minds, then the URL is second.

2. In the display of the previewURL, can it be browser blue instead of green?  Otherwise ppl may not understand it is clickable.

3. Is the URL clickable?

File: platform-api/che-core-api-core/src/main/java/org/eclipse/che/api/core/model/machine/Command.java (hunk 3 above)
Comment (vzhukovskii) [inline platform-api/che-core-api-core/src/main/java/org/eclipse/che/api/core/model/machine/Command.java]: I don't like either. But according to issue we need to be able to store preview url for each command.

Comment (garagatyi) [inline platform-api/che-core-api-core/src/main/java/org/eclipse/che/api/core/model/machine/Command.java]: Preview url as a string is useless. [Here](https://codenvy.slack.com/archives/general/p1451318006000395) Tyler proposed another scheme. If we are OK to implement this, we need also realize how to set file: preview object.
Preview object as  string won't allow us to implement proposed strategies.

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/model/impl/CommandImpl.java [hunk 4]
```diff
@@ -65,17 +67,23 @@ public void setType(String type) {
...
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```
Comment (evoevodin) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/model/impl/CommandImpl.java]: Violates [Lisokv substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle) from _Effective java 2nd ed item 8_

Comment (vzhukovskii) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/model/impl/CommandImpl.java]: It's an artifact after generation equals method via intellij idea, reverted to original comparison algorithm.

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/model/impl/CommandImpl.java [hunk 5]
```diff
@@ -73,9 +84,10 @@ public boolean equals(Object obj) {
             return false;
         }
         final CommandImpl command = (CommandImpl)obj;
-        return Objects.equals(name, command.name) &&
-               Objects.equals(commandLine, command.commandLine) &&
-               Objects.equals(type, command.type);
+        return Objects.equals(getName(), command.getName()) &&
```
Comment (evoevodin) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/model/impl/CommandImpl.java]: Previous version of the equals was pretty ok, you could just add a new line of `Objects.equals` and that's it, why did you add getters here?

Comment (vzhukovskii) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/model/impl/CommandImpl.java]: Haven't looked on it. Fixed, thanks.

**Comment by vzhukovskii** [comment]
@TylerJewell here you go


>Can we move the preview URL portion of the command popup to be the last entry, after "Command line"? The command line is first in people's minds, then the URL is second.
Done

>In the display of the previewURL, can it be browser blue instead of green? Otherwise ppl may not understand it is clickable.
Color has been changed

>Is the URL clickable?
Yes, it's simple anchor element

**Comment by vzhukovskii** [comment]
Updated state:
